posthog/hogql/database/schema/person_distinct_id_overrides.py

```diff
@@ -0,0 +1,92 @@
+from typing import Dict, List
+from posthog.hogql.ast import SelectQuery
+from posthog.hogql.context import HogQLContext
+
+from posthog.hogql.database.argmax import argmax_select
+from posthog.hogql.database.models import (
+    Table,
+    IntegerDatabaseField,
+    StringDatabaseField,
+    BooleanDatabaseField,
+    LazyJoin,
+    LazyTable,
+    FieldOrTable,
+)
+from posthog.hogql.database.schema.persons import join_with_persons_table
+from posthog.hogql.errors import HogQLException
+from posthog.schema import HogQLQueryModifiers
+
+PERSON_DISTINCT_ID_OVERRIDES_FIELDS = {
+    "team_id": IntegerDatabaseField(name="team_id"),
+    "distinct_id": StringDatabaseField(name="distinct_id"),
+    "person_id": StringDatabaseField(name="person_id"),
+    "person": LazyJoin(
+        from_field=["person_id"],
+        join_table="persons",
+        join_function=join_with_persons_table,
+    ),
+}
+
+
+def select_from_person_distinct_id_overrides_table(requested_fields: Dict[str, List[str | int]]):
+    # Always include "person_id", as it's the key we use to make further joins, and it'd be great if it's available
+    if "person_id" not in requested_fields:
+        requested_fields = {**requested_fields, "person_id": ["person_id"]}
+    return argmax_select(
+        table_name="raw_person_distinct_id_overrides",
+        select_fields=requested_fields,
+        group_fields=["distinct_id"],
+        argmax_field="version",
+        deleted_field="is_deleted",
+    )
+
+
+def join_with_person_distinct_id_overrides_table(
+    from_table: str,
+    to_table: str,
+    requested_fields: Dict[str, List[str]],
+    context: HogQLContext,
+    node: SelectQuery,
+):
+    from posthog.hogql import ast
+
+    if not requested_fields:
+        raise HogQLException("No fields requested from person_distinct_id_overrides")
+    join_expr = ast.JoinExpr(table=select_from_person_distinct_id_overrides_table(requested_fields))
+    join_expr.join_type = "LEFT OUTER JOIN"
```

Comment:
Assuming we are able to stay reasonably on top of merges, something that could be interesting to check out here would be using an `ANY` merge in place of the `argMax` to see if that has any performance impact. I suspect it wouldn't be worth it since it'd increase cardinality of the RHS table and has the potential to create nondeterminism headaches if there are lots of rows with duplicate keys, but hard to say that for sure without trying.